Content.Server/GameObjects/Components/Items/ToysComponent.cs

```diff
@@ -24,36 +24,54 @@ public class ToysComponent : Component, IActivate, IUse, ILand
         [ViewVariables]
         [DataField("toySqueak")]
         public string _soundCollectionName = "ToySqueak";
+        private EmitSoundOnLandComponent? _emitSoundOnLand = default;
+        private EmitSoundOnUseComponent? _emitSoundOnUse = default;
 
-        public void Squeak()
+        public override void Initialize()
         {
-            PlaySqueakEffect();
+            base.Initialize();
+            Owner.EnsureComponent(out EmitSoundOnLandComponent emitSoundOnLand);
+            Owner.EnsureComponent(out EmitSoundOnUseComponent emitSoundOnUse);
+            _emitSoundOnLand = emitSoundOnLand;
+            _emitSoundOnUse = emitSoundOnUse;
```

Comment:
I'm not sure if this is the proper way to use and setup Components, I'm open to comments if the code should be improved. But the functionality does work.